product_pricelist_multiwebsite_support/models/product_hook_website_price.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.http import request
+from odoo import models, fields, api
+from odoo.tools import float_is_zero
+from odoo.addons.website_sale.models.product import Product
+
+# Monkey Patching
+# Overwrite the original _website_price in wabsite_sale
+# i.e. https://github.com/odoo/odoo/blob/10.0/addons/website_sale/models
+# /product.py#L196-L213
+def _website_price(self):
+    qty = self._context.get('quantity', 1.0)
+    partner = self.env.user.partner_id
+    current_website = self.env['website'].get_current_website()
+    pricelist = current_website.get_current_pricelist()
+    company_id = current_website.company_id
+
+    # Check the request user
+    # If the user is login-ed, apply the pricelist of the user
+    if request.env.user != request.website.user_id and request.uid <> \
+            partner.id:
+        partner = self.env["res.users"].browse(request.uid).partner_id
+        pricelist = partner.property_product_pricelist
```

Comment:
Please mark the changed part with `[QTL]` or something so that it is obvious which part *we* have modified.

Reply:
@yostashiro  I have modified the comment